server/lib/tuist_web/controllers/slack_oauth_controller.ex

```diff
@@ -0,0 +1,76 @@
+defmodule TuistWeb.SlackOAuthController do
+  @moduledoc """
+  Controller for handling Slack OAuth flow.
+  """
+
+  use TuistWeb, :controller
+
+  alias Tuist.Accounts
+  alias Tuist.Environment
+  alias Tuist.Slack
+  alias Tuist.Slack.Client, as: SlackClient
+  alias TuistWeb.Errors.BadRequestError
+
+  @slack_scopes "chat:write,chat:write.public,channels:read,groups:read"
+
+  def callback(conn, %{"code" => code, "state" => state_token})
+      when is_binary(code) and code != "" and is_binary(state_token) do
+    with {:ok, account_id} <- verify_state_token(state_token),
+         {:ok, account} <- Accounts.get_account_by_id(account_id),
+         {:ok, token_data} <- SlackClient.exchange_code_for_token(code, slack_redirect_uri()),
+         {:ok, _installation} <- create_installation(account, token_data) do
+      redirect(conn, to: ~p"/#{account.name}/integrations")
+    else
+      {:error, :invalid_state_token} ->
+        raise BadRequestError, dgettext("dashboard", "Invalid authorization request. Please try again.")
```

Comment:
The drawback of using modular locale files to avoid conflicts is that we need to ensure we respect that in the domains, so I suggest we use something else here, maybe `dashboard_slack`